### PR TITLE
feat(actions): refactoring + improvement of github actions

### LINKED ISSUE #21275: Update labelling guidelines or workflow
### Description
In our PR template and labeling guidelines, we emphasize the importance of applying the correct label to each pull request (PR). This practice is crucial for enhancing our PR taxonomies and categorization. However, a challenge arises when external contributors attempt to add labels to their PRs, as they do not have the necessary permissions. To alleviate this potential source of confusion, we could automate the addition of the [external-contributor](https://github.com/MetaMask/metamask-extension/labels/external-contributor) label to PRs originating outside of the MetaMask organization. We could also provide more clarity within the PR template and labeling guidelines.

### Technical Details
- Automate adding [external-contributor](https://github.com/MetaMask/metamask-extension/labels/external-contributor) label to PRs outside of the MetaMask organization
- Update [PR template](https://github.com/MetaMask/metamask-extension/blob/8013e24c1c5683e23825d345eac0c9776527ad72/.github/pull-request-template.md#L38):  "I’ve applied the correct labels to the PR (refer to labeling guidelines." to "I’ve applied the correct labels to the PR (refer to labeling guidelines). Not required for external contributors"
- Update [PR Labeling Guidelines](https://github.com/MetaMask/metamask-extension/blob/8013e24c1c5683e23825d345eac0c9776527ad72/.github/guidelines/LABELING_GUIDELINES.md#L7): "**External Contributors**: PRs from contributors outside the organization must have the `external-contributor` label." to "**External Contributors**: PRs submitted by contributors who are not part of the organization will be automatically labeled with `external-contributor`."

### Acceptance Criteria
- The workflow correctly identifies and labels PRs from external contributors.
- The external-contributor label is automatically added to PRs from external contributors.
- PRs from MetaMask organization members remain unaffected.
- It is clear to contributors outside of the MetaMask organization that they don't have to add labels to their PRs

### PR DESCRIPTION
**Pull request by gauthierpetetin**
## **Description**
The purpose for these changes is to refactor the github action that checks issues template and adds labels in order to make it more generic and apply it to PRs as well.
Some code duplication has been eliminated as part of this refactoring as well.

[Same PR for Mobile.](https://github.com/MetaMask/metamask-mobile/pull/7567)

## **Manual testing steps**

1. Go on this [test repo](https://github.com/gauthierpetetin-test/repo_test/)
2. Open a PR
3. Wait for ~30s and `external-contributor` label shall appear on the PR you just opened
4. Edit the PR by deleting the whole description
5. Wait for ~30s and the `INVALID-PR-TEMPLATE` shall be added

## **Screenshots/Recordings**

https://recordit.co/5sH8eRHVmZ

### **Before**

NA

### **After**

NA

## **Related issues**

Fixes https://github.com/MetaMask/metamask-extension/issues/21275

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by georgewrmarshall — COMMENTED** [review]
This is excellent! As you noted in slack [my test PR failed](https://github.com/gauthierpetetin-test/repo_test/pull/153) so it may need some revision. My experience with GitHub workflows and yml scripts is limited but from a general code perspective it looks good. I would suggest getting a second reviewer familiar with our workflow scripts 

Screenshot 2023-10-23 at 12 08 59 PM

**Comment by gauthierpetetin** [comment]
Hi @georgewrmarshall , here's a [fix](https://github.com/MetaMask/metamask-extension/pull/21497/commits/c9731e6c210b366452f6ceace55566da04f3f6eb) to your comment regarding PR from forked repo.
- When PR is opened from a forked repo, by default, the workflow runs in the context of the forked repo which doesn’t have the LABEL_TOKEN secret defined, and results in an error.
- Replacing `pull_request` by `pull_request_target` allows the workflow to run in the context of the base repo rather than the forked repo. So it has access to LABEL_TOKEN secret and it works as expected.

File: .github/scripts/shared/label.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+export interface Label {
+  name: string;
+  color: string;
+  description: string;
+}
+
+export const externalContributorLabel: Label = {
+  name: 'external-contributor',
+  color: 'B60205',
```
[the remark this answers is not in the document]
Comment (gauthierpetetin) [reply .github/scripts/shared/label.ts]: Fixed in this [commit](https://github.com/MetaMask/metamask-extension/pull/21497/commits/aabd64c41917b818b324a295855863d0641f98d7)

**Review by georgewrmarshall — APPROVED** [review]
LGTM. Fantastic work 💯  Tested labelling on test repo working well! https://github.com/gauthierpetetin-test/repo_test/pull/159

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**